✨ Add new arbitrary called `uniqueArray` (#2616)
dubzzz/fast-check: documentation/Arbitraries.md

>
> All the values in the array are unique given the provided `selector` and `comparator` functions.
> Two values `a` and `b` are considered equal and thus would not be selected together in the same generated array if and only if `comparator(selector(a), selector(b))` is `true`.
>
> If not specified `selector` defaults to the identity function and `comparator` defaults to `SameValue` aka `Object.is`.
>
> For performance reasons, we highly discourage the use of a fully custom `comparator` and recommend to rely on a custom `selector` function whenever possible.
> Such custom `comparator` — outside of provided ones — cannot be properly optimized and thus includes a potentially huge performance penalty mostly impacted large arrays.

*&#8195;Signatures*

- `fc.set(arb)`
- `fc.set(arb, {minLength?, maxLength?, selector?, comparator?, size?})`

*&#8195;with:*

- `arb` — _arbitrary instance responsible to generate values_
- `minLength?` — default: `0` — _minimal length (included)_
- `maxLength?` — default: `0x7fffffff` [more](#size-explained) — _maximal length (included)_
- `selector?` — default: `v => v` — _project the generated value in order to compare it_

> - `comparator?` — default: `SameValue` — _compare two projected values and returs `true` if the to projected values should be considered equal_

Author: ```suggestion
- `comparator?` — default: `SameValue` — _compare two projected values and returns `true` whenever the projected values should be considered equal_
```
